src/riscvemu.c

```diff
@@ -583,12 +583,16 @@ void virt_machine_run(VirtMachine *m)
 #ifndef _WIN32
         if (m->console_dev && FD_ISSET(stdin_fd, &rfds)) {
             uint8_t buf[128];
-            int ret, len;
-            len = virtio_console_get_write_len(m->console_dev);
-            len = min_int(len, sizeof(buf));
+            int ret, len = sizeof(buf), ulen = uart_can_rx(m);
+	    if (virtio_console_can_write_data(m->console_dev))
```

Comment:
Please use 4 spaces instead of one tab character. This line is mixing tabs and spaces.